src/cpp/ops/ops.cpp

```diff
@@ -235,4 +237,209 @@ serialize(std::shared_ptr<assembler_state> state,
   return ret;
 }
 
+
+uint32_t op_deserializer::numlabel = 0;
+
+uint32_t
+isa_op_deserializer::
+deserialize(asm_writer& writer, std::shared_ptr<disassembler_state> state, const char* data)
+{
+  std::vector<std::string> result;
+  //1 byte opcode and 1 byte pad
+  uint32_t size = 2;
+  uint32_t tile = 0;
+
+  for (const auto& arg : m_opcode->get_args()) {
+    uint32_t len = arg.get_width() / byte_to_bits;  // convert bits to byte
+    uint32_t val = get_arg_val(data + size, len);
+    size += len;
+    switch (arg.get_type()) {
```

Comment:
1. Please move the code for each case to a dedicated suitably named function. 
2. Replace checks like arg.get_name() == "actor_id" with something like is_actor(arg) where is actor() is a inline function
3. Document the logic of if/else-if/else for code readability